### PR TITLE
2155 modals

### PR DESCRIPTION
**Pull request by estellecomment**
#2155

### CODE REVIEW

File: static/js/controllers/inbox.js [hunk 1]
```diff
@@ -661,28 +662,32 @@ var feedback = require('../modules/feedback'),
         $rootScope.$broadcast.apply($rootScope, arguments);
       };
 
-      var docToDeleteId;
-
-      $scope.deleteDoc = function(id) {
-        $('#delete-confirm').modal('show');
-        docToDeleteId = id;
+      // TODO promisify DeleteDoc
```
Comment (garethbowen) [inline static/js/controllers/inbox.js]: Don't commit TODOs - raise an issue instead. The only time I see TODOs is when I'm busy doing something else, whereas I see issues when I'm looking for something to work on.

File: static/js/services/modal.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
+(function () {
+
+  'use strict';
+
+  /**
+   * Creates a yes/no modal and gives back the user's decision in promise.
```
Comment (garethbowen) [inline static/js/services/modal.js]: Is this only for yes/no modals?

File: static/js/services/modal.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+   * };
+   */
+  var inboxServices = angular.module('inboxServices');
+
+  inboxServices.factory('Modal', ['$uibModal',
+    function($uibModal) {
+      return function(templateUrl, processingFunction) {
+        var modalInstance = $uibModal.open({
+          templateUrl: templateUrl,
+          controller: 'ModalCtrl',
```
Comment (garethbowen) [inline static/js/services/modal.js]: As discussed, we need one controller per modal, or ModalCtrl will end up a mess with all modals fields and functions in its $scope.

File: templates/modals/alert.html [hunk 4]
```diff
@@ -1,18 +1,17 @@
...
-      <div class="modal-body">
-        <p>{{'confirm.delete' | translate}}</p>
-      </div>
-      <div class="modal-footer">
-        <p class="note"></p>
-        <a class="btn cancel" data-dismiss="modal">{{'Cancel' | translate}}</a>
-        <a class="btn submit btn-danger" ng-click="deleteDocConfirm()" ng-attr-data-working-label="{{'Deleting' | translate}}">{{'Delete' | translate}}</a>
-      </div>
-    </div>
+<div id="delete-confirm" role="document">
```
Comment (garethbowen) [inline templates/modals/alert.html]: I don't think we need this ID any more as it was used to show/hide the modal.

File: templates/modals/navigation_confirm.html [hunk 5]
```diff
@@ -1,17 +1,13 @@
...
-      </div>
-      <div class="modal-body">
-        <p>{{'confirm.destructive.navigation.forms' | translate}}</p>
-      </div>
-      <div class="modal-footer">
-        <a class="btn cancel" data-dismiss="modal">{{'no' | translate}}</a>
-        <a class="btn submit btn-danger" ng-click="cancelConfirm()">{{'yes' | translate}}</a>
-      </div>
-    </div>
+<div id="navigation-confirm" role="document">
```
Comment (garethbowen) [inline templates/modals/navigation_confirm.html]: As above - id not required.

File: tests/karma/unit/controllers/modal.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        scope.$apply(); // needed to resolve the promises
+        chai.assert(!spyUibModalInstance.close.called, 'Should not close modal when processing error');
+        chai.assert(!spyUibModalInstance.dismiss.called, 'Should not dismiss modal when processing error');
+        chai.assert(scope.error, 'Should be displaying error when processing error');
+        chai.assert(!scope.processing, 'Should not be processing when processing is done');
+        done();
+      });
+    });
+  });
+});
```
Comment (garethbowen) [inline tests/karma/unit/controllers/modal.js]: Nice test coverage!

File: static/js/controllers/inbox.js (hunk 1 above)
Comment (estellecomment) [inline static/js/controllers/inbox.js]: good point. Filed #2194.

**Pull request merged**